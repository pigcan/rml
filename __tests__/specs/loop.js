
const MLTransformer = require('../../src/Transformer');

describe('loop', () => {
  it('support simple', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:for="{{items}}">',
      '{{item}}',
      '</div>',
      '<div>{{i}}</div>',
      '</div>',
    ].join('\n')).transform((err, code) => {
      expect(code).toMatchSnapshot();
      done();
    });
  });

  it('support root simple', (done) => {
    new MLTransformer(`
<view r:for="z">
<block>1</block>
</view>
`.trim())
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });

  it('support nested', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:for="{{items}}" r:for-item="outer">',
      '<div r:for="{{outer}}">',
      '{{item}}',
      '</div>',
      '</div>',
      '</div>',
    ].join('\n'))
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });

  it('support item', (done) => {
    new MLTransformer([
      '<div>',
      '<div r:for="{{items}}" r:for-index="i" r:for-item="t" r:key="u">',
      '{{i}} - {{t}} - {{z}}',
      '</div>',
      '<div>{{i}}</div>',
      '</div>',
    ].join('\n'))
      .transform((err, code) => {
        expect(code).toMatchSnapshot();
        done();
      });
  });
});
