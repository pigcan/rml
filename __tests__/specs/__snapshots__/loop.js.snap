// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loop support item 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      ((data.items) || []).map((t, i) => {
        return (
          <div
            key = {t.u}
          >
            {[(i), ' - ', (t), ' - ', (data.z)]}
          </div>
        );
      })
      }
      <div>
        {(data.i)}
      </div>
    </div>
  );
};"
`;

exports[`loop support nested 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      ((data.items) || []).map((outer, index) => {
        return (
          <div>
            {
            ((outer) || []).map((item, index) => {
              return (
                <div>
                  {(item)}
                </div>
              );
            })
            }
          </div>
        );
      })
      }
    </div>
  );
};"
`;

exports[`loop support root simple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    ('z' || []).map((item, index) => {
      return (
        <view>
          {'1'}
        </view>
      );
    })
  );
};"
`;

exports[`loop support simple 1`] = `
"import React from 'react';

export default function render(data) {
  return (
    <div>
      {
      ((data.items) || []).map((item, index) => {
        return (
          <div>
            {(item)}
          </div>
        );
      })
      }
      <div>
        {(data.i)}
      </div>
    </div>
  );
};"
`;
